pgjdbc/src/main/java/org/postgresql/core/BaseQueryKey.java

```diff
@@ -0,0 +1,47 @@
+package org.postgresql.core;
+
+public class BaseQueryKey {
+  public final String sql;
+  public final boolean isParameterized;
+  public final boolean escapeProcessing;
+
+  public BaseQueryKey(String sql, boolean isParameterized, boolean escapeProcessing) {
+    this.sql = sql;
+    this.isParameterized = isParameterized;
+    this.escapeProcessing = escapeProcessing;
+  }
+
+  @Override
+  public String toString() {
+    return sql;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
```

Comment:
is there an optimizer reason why not to check this first ? Or do you expect the first == clause to be more frequent ?